### PR TITLE
RJD-1336/fix_request_speed_change_throws

### PR DESCRIPTION
**Pull request by robomic**
## Abstract

Please refer to [this issue](https://github.com/tier4/scenario_simulator_v2/issues/1393).

## Details

1. Inversion of order in if statements

`  if (!continuous && isTargetSpeedReached(target_speed)) `  --->
 ` if (isTargetSpeedReached(target_speed) && !continuous)`
this forces calculation of relative_target_speed, which throws immediately, if invalid.
2. Numerous `const` qualifiers added.
3. A single `cpp_mock_scenario` has been modified in order to update `other_status` variable.
## References
[issue](https://github.com/tier4/scenario_simulator_v2/issues/1393)
[jira ticket](https://tier4.atlassian.net/browse/RJD-1336)

### CODE REVIEW

**Comment by robomic** [comment]
[Regression results](https://tier4.atlassian.net/browse/RJD-1336?focusedCommentId=190062): OK

File: mock/cpp_mock_scenarios/src/speed_planning/request_speed_change_relative.cpp [hunk 1]
```diff
@@ -67,6 +67,7 @@ class RequestSpeedChangeRelativeScenario : public cpp_mock_scenarios::CppScenari
         34741, 10.0, 0.0, api_.getHdmapUtils()),
       getVehicleParameters());
     api_.setLinearVelocity("front", 3);
+    api_.updateFrame();
```
Comment (hakuturu583) [inline mock/cpp_mock_scenarios/src/speed_planning/request_speed_change_relative.cpp:70]: This function is intended to be called on the base type and is not expected to be called within scenarios. Access rights modifications will be implemented in a separate PR, so no changes are needed in this PR.

File: simulation/traffic_simulator/src/entity/entity_base.cpp [hunk 2]
```diff
@@ -329,19 +330,20 @@ void EntityBase::requestSpeedChangeWithConstantAcceleration(
...
-          if (diff > 0) {
+          if (diff > 0.1) {
             setAccelerationLimit(std::abs(acceleration));
             return false;
           }
-          if (diff < 0) {
+          if (diff < 0.1) {
             setDecelerationLimit(std::abs(acceleration));
             return false;
           }
```
Comment (hakuturu583) [inline simulation/traffic_simulator/src/entity/entity_base.cpp:349]: Use constexpr for hard coded parameter 0.1

E.g 

```
constexpr double speed_difference_threshold = 0.1;
if (std::abs(diff) <= speed_difference_threshold) {
    return true;
}
if (diff > speed_difference_threshold) {
    setAccelerationLimit(std::abs(acceleration));
    return false;
}
```

**Review by hakuturu583 — CHANGES_REQUESTED** [review]

File: mock/cpp_mock_scenarios/src/speed_planning/request_speed_change_relative.cpp (hunk 1 above)
Comment (robomic) [reply mock/cpp_mock_scenarios/src/speed_planning/request_speed_change_relative.cpp:70]: This line is needed in order to pass this particular test. Without it `other_status_` variable is not updated and does not contain other entities. Empty `other_status_` results in a throw when calculating the absolute speed from `speed_change::RelativeTargetSpeed` variable.

Comment (hakuturu583) [reply mock/cpp_mock_scenarios/src/speed_planning/request_speed_change_relative.cpp:70]: If so, you should wait for one frame to elapse before making a decision, and should not implement the decision on the spot.

**Review by hakuturu583 — CHANGES_REQUESTED** [review]

File: mock/cpp_mock_scenarios/src/speed_planning/request_speed_change_relative.cpp (hunk 1 above)
Comment (robomic) [reply mock/cpp_mock_scenarios/src/speed_planning/request_speed_change_relative.cpp]: i have moved `requestSpeedChange` to `onUpdate`

**Review by hakuturu583 — APPROVED** [review]
LGTM
